corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js

```diff
@@ -910,21 +910,48 @@ hqDefine("cloudcare/js/form_entry/entries", function () {
         };
 
         self.file = ko.observable();
-
+        self.extensionsMap = {
+            "image/*,.pdf": ["jpg", "jpeg", "png", "pdf"],
+            "audio/*": ["3ga","mp3", "wav", "amr", "qcp","ogg"],
+            "video/*": ["3gpp", "3gp", "3gp2", "3g2", "mp4","mpg4", "mpeg4", "m4v", "mpg", "mpeg"],
+        };
     }
     FileEntry.prototype = Object.create(EntrySingleAnswer.prototype);
     FileEntry.prototype.constructor = EntrySingleAnswer;
+    FileEntry.prototype.onPreProcess = function (newValue) {
+        var self = this;
+        if (newValue !== constants.NO_ANSWER && newValue !== "") {
+            const ext = newValue.slice(newValue.lastIndexOf(".") + 1);
+            const acceptedExts = self.extensionsMap[self.accept];
+            if (!acceptedExts.includes(ext.toLowerCase())) {
+                self.question.error(gettext("Invalid file type chosen. Please select a valid multimedia file."));
+                return;
+            }
+            this.answer(newValue.replace(constants.FILE_PREFIX, ""));
+        }
+    };
     FileEntry.prototype.onAnswerChange = function (newValue) {
         var self = this;
-        if (newValue !== constants.NO_ANSWER) {
+        // file has already been assigned a unique id and another request should not be sent to formplayer
+        if (self.answer() && self.answer().match(/^(\w{8}-)(\w{4}-){3}(\w{12})\.\w{3,4}$/)) {
+            return;
+        }
+        if (newValue !== constants.NO_ANSWER && newValue !== "") {
             var $input = $('#' + self.entryId);
-            self.answer(newValue.replace(constants.FILE_PREFIX, ""));
             self.file($input[0].files[0]);
+            if (self.file().size > 3000000) {
```

Comment:
This would also be a good place for a comment referencing the relevant formplayer code. Ideally formplayer should also have comments referencing the js (I think that's typical for constants shared between formplayer & web apps, @shubham1g5 correct me if it's not).

We're likely to change this limit and want to make sure it's done everywhere.

Reply:
updated comments here https://github.com/dimagi/commcare-hq/pull/32634/commits/0834bc61f5d467426f996a207084f6178cb90e4b to make sure changes are made in js and FP.